br/pkg/lightning/web/progress.go

```diff
@@ -108,8 +108,14 @@ type taskProgress struct {
 	checkpoints checkpointsMap
 }
 
-var currentProgress = taskProgress{
-	checkpoints: makeCheckpointsMap(),
+var currentProgress *taskProgress
+
+// InitCurrentProgress init current progress struct on demand.
+// NOTE: this call is not thread safe, so it should only be inited once at the very beginning of progress start.
```

Comment:
It seems it is not very necessary to init `currentProgress` on demand since the `checkpointsMap` has almost no memory cost. If we keep the original init fashion, some panic can be avoided and it is also thread safe.

Reply:
Yes. I deliberately to not init the `currentProgress` so it will cause panic if any `BroadcastProcss` operations are accidentally called when status-port is disable.

Reply:
`handleProgressTask` and `handleProgressTable` may have `BroadcastProcss` operations if sending SIGUSR1 to enable the status port. Do we need a check there?

Reply:
NO. I have changed the implement. In the newest logic, the global atomic variable `progressEnabled` is set at process starting and won't change anymore. Each call of `BroadcastXXX` will first check if this variable is enable, if not just do nothing. So the caller logic can remain unchanged.